src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,119 @@
+import java.io.PrintStream;
+import java.lang.reflect.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReflectionTest {
+    private static final Logger logger = LoggerFactory.getLogger(ReflectionTest.class);
+    private Class<Car> carClass;
+
+    final String NAME = "메르세데스 벤츠";
+    final int PRICE = 1000;
+
+    @BeforeEach
+    void setUp() {
+        carClass = Car.class;
+    }
+
+    @Test
+    @DisplayName("요구사항1. Car 클래스의 모든 필드, 생성자, 메소드 정보를 출력한다.")
+    void showClass() {
+        List<String> fields = convertArrayToListWithMapping(carClass.getDeclaredFields(), Field::getName);
+        List<String> constructor = convertArrayToListWithMapping(carClass.getDeclaredConstructors(), Constructor::getName);
+        List<String> methods = convertArrayToListWithMapping(carClass.getDeclaredMethods(), Method::getName);
+
+        assertThat(fields).containsOnly("name", "price");
+        assertThat(constructor).containsOnly("persistence.study.Car", "persistence.study.Car");
+        assertThat(methods).containsOnly("testGetPrice", "printView", "testGetName", "getName", "getPrice");
+    }
+
+    private <T, R> List<R> convertArrayToListWithMapping(T[] array, Function<T, R> map) {
+        return Arrays.stream(array).map(map).collect(Collectors.toList());
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메소드 실행")
+    void testMethodRun() throws InvocationTargetException, IllegalAccessException {
+        carClass = Car.class;
+        Car car = new Car();
+        Method[] methods = carClass.getMethods();
+        for (Method method : methods) {
+            isStartWithTest(car, method);
+        }
+    }
+
+    private void isStartWithTest(final Car car, final Method method) throws IllegalAccessException, InvocationTargetException {
+        if (method.getName().startsWith("test")) {
+            String invoke = (String) method.invoke(car);
+            assertThat(invoke).startsWith("test : ");
+        }
+    }
+
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메소드 실행")
+    void testAnnotationMethodRun() throws InvocationTargetException, IllegalAccessException {
+        Car car = new Car();
+        OutputStream out = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(out));
+
+        List<Method> methodWithPrintView = Arrays.stream(carClass.getMethods())
+                .filter(method -> method.isAnnotationPresent(PrintView.class)).collect(Collectors.toList());
+        for (Method method : methodWithPrintView) {
+            method.invoke(car);
+            assertThat(out.toString().trim()).isEqualTo("자동차 정보를 출력 합니다.");
+        }
+    }
+
+    @Test
+    @DisplayName("private field에 값 할당")
+    void privateFieldAccess() throws IllegalAccessException, NoSuchFieldException, NoSuchMethodException, InvocationTargetException, InstantiationException {
```

Comment:
사소하지만 최상위 예외를 던지도록 해보는건 어떠실까요? 😄
저는 개인적으로 테스트 코드에 한해서만 가독성을 위해 아래 처럼 사용하곤 합니다!
```suggestion
    void privateFieldAccess() throws Exception {
```